flask/packages.py

```diff
@@ -0,0 +1,91 @@
+from app import app
+from mongo import db
+from flask import render_template
+from flask import request, make_response, jsonify
+from datetime import datetime
+from auth import generate_uuid
+
+@app.route('/packages', methods=["GET"])
+def search_packages():
+    query = request.args.get('query')
+    packages = db.packages.find(
+        {
+            "$or": [
+                {"name": {"$regex": query}},
+                {"tags": {"$in": [query]}},
+                {"description": {"$regex": query}},
+            ]
+        }
+    )
+
+    return jsonify([package for package in packages])
+
+@app.route("/packages", methods=["POST"])
+def upload(username):
```

Comment:
``username`` here should be degined in the route as well like ``/packages/<username>``

Reply:
we had a discussion with milan regarding this so we wanted to keep the upload to be independent of the username.